src/plugins/auto/auto_executable_network.cpp

```diff
@@ -237,7 +237,7 @@ IE::Parameter AutoExecutableNetwork::GetMetric(const std::string& name) const {
         if (_autoSchedule->_pCTPUTLoadContext) {
             std::vector<std::string> exeDevices = {};
             std::lock_guard<std::mutex> lock(_autoSContext->_confMutex);
-            for (auto n : _autoSContext->_devicePriorities) {
+            for (auto & n : _autoSContext->_devicePriorities) {
```

Comment:
`auto&&`

Reply:
do we have a reason && is a must? if no, I prefer const auto&

Reply:
https://stackoverflow.com/questions/29859796   Shall we use auto const & ?

Reply:
From my understanding, `auto &` works well if it doesn't bind to a temporary object.
`const auto&` could do that, while it doesn't allow to change the binding object.
`auto&&` could bind to both l-value reference and r-value reference, and also allow us to modify the value of the object.

Reply:
If we don't want to modify element value, `const auto&` is a good choice.
`auto&&` maybe more common for lvalue or rvalue reference, but if we don't want to modify its value, `const auto&` is recommended.